ansible_collections/f5networks/f5_modules/plugins/modules/bigip_device_info.py

```diff
@@ -8866,6 +8920,57 @@ def read_collection_from_device(self):
         return result
 
 
+class DOParameters(BaseParameters):
+    api_map = {
+    }
+
+    returnables = [
+
+    ]
+
+
+class DOFactManager(BaseManager):
+    def __init__(self, *args, **kwargs):
+        self.client = kwargs.get('client', None)
+        self.module = kwargs.get('module', None)
+        super(DOFactManager, self).__init__(**kwargs)
+        self.want = self.module.params
+
+    def exec_module(self):
+        facts = self._exec_module()
+        result = dict(do_config=facts)
+        return result
+
+    def _exec_module(self):
+        results = []
+        facts = self.read_facts()
+        return facts
+
+    def read_facts(self):
+        results = []
+        collection = self.read_collection_from_device()
+        return collection
+
+    def read_collection_from_device(self):
+        uri = "https://{0}:{1}/mgmt/shared/declarative-onboarding/config".format(
+            self.client.provider['server'],
+            self.client.provider['server_port'],
+        )
+        resp = self.client.api.get(uri)
+        try:
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+
+        if resp.status not in [200, 201] or 'code' in response and response['code'] not in [200, 201]:
+            raise F5ModuleError(resp.content)
+
+        if 'parsed' not in response[0]:
+            return []
+        result = response[0]
+        return result
```

Comment:
also under what key declaration exists in the API?

Reply:
I don't understand the question... By the way, this is an amalgam of AS3, DO, and TS checkins... I just merged them all together.

Reply:
see above comment

Reply:
new output looks like - 
```
"ts_config": {
            "declaration": {
                "My_Consumer": {
                    "allowSelfSignedCert": false,
                    "class": "Telemetry_Consumer",
                    "enable": true,
                    "format": "default",
                    "host": "192.0.2.1",
```
```
"do_config": {
            "declaration": {
                "Common": {
                    "Authentication": {
                        "enabledSourceType": "local",
                        "fallback": false,
                        "remoteUsersDefaults": {
                            "partitionAccess
```
```
"as3_config": {
            "declaration": {
                "Sample_01": {
                    "A1": {
                        "class": "Application",
                        "serviceMain": {
```